src/main/java/com/petstore/petstoreRest/service/impl/UserServiceImpl.java

```diff
@@ -1,79 +1,79 @@
 package com.petstore.petstoreRest.service.impl;
 
 import com.petstore.petstoreRest.dto.UserDTO;
-import com.petstore.petstoreRest.entity.User;
+import com.petstore.petstoreRest.exception.UserAlreadyExistException;
 import com.petstore.petstoreRest.mapper.UserMapper;
 import com.petstore.petstoreRest.repository.UserRepository;
 import com.petstore.petstoreRest.service.UserService;
+import jakarta.persistence.EntityNotFoundException;
 import lombok.RequiredArgsConstructor;
-import org.springframework.http.HttpStatus;
+import org.springframework.dao.DataAccessException;
 import org.springframework.stereotype.Service;
-import org.springframework.web.server.ResponseStatusException;
 
 import java.util.List;
-import java.util.Optional;
 
+/**
+ * This implementation of {@link UserService} provides methods for managing user accounts.
+ */
 @Service
 @RequiredArgsConstructor
 public class UserServiceImpl implements UserService {
 
     private final UserMapper userMapper;
+
     private final UserRepository userRepository;
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
     public UserDTO findByUsername(String username) {
         return userMapper.toDTO(userRepository.findByUserName(username)
-                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND)));
-        //TODO: change to other exception
-        //                .orElseThrow(() -> new EntityNotFoundException("User with username \"" + username + "\" not found")));
+                .orElseThrow(() -> new EntityNotFoundException("User with username \"" + username + "\" not found")));
```

Comment:
please use String.format() static method when you construct exception message

Reply:
Changed